selfdrive/ui/qt/widgets/controls.cc

```diff
@@ -85,3 +87,14 @@ ButtonControl::ButtonControl(const QString &title, const QString &text, const QS
   QObject::connect(&btn, &QPushButton::released, this, &ButtonControl::released);
   hlayout->addWidget(&btn);
 }
+
+void ElidedLabel::paintEvent(QPaintEvent *event) {
+  QFrame::paintEvent(event);
+
+  QPainter painter(this);
```

Comment:
can't we just make a custom QLabel and override setText?

Reply:
We may not know the width of qlabel when calling settext

Reply:
we can use the showEvent